runtime/compiler/control/CompilationThread.cpp

```diff
@@ -11637,6 +11637,11 @@ TR::CompilationInfo::storeAOTInSharedCache(
                 * to send two different buffers to store in cache and also as no one queries either code data / metadata for
                 * method, clean up the share classs cache API.
                 */
+               TR_ASSERT(comp, "The AOT compilation that succeeded must have a compilation object.");
```

Comment:
Given that the program will fail with a core dump if these ASSERTs are false I think it would be better for them to be TR_ASSERT_FATAL to ge a good debug message (if they are worth checking).

Reply:
The assert on comp comes too late because the code above does `if (!comp->getOption(TR_DisableAOTBytesCompression) && !comp->isRemoteCompilation())`
The caller of `storeAOTInSharedCache` already makes sure that comp is not null, so there is no need for another assert.